crates/egui/src/context.rs

```diff
@@ -1200,12 +1199,18 @@ impl Context {
     pub fn read_response(&self, id: Id) -> Option<Response> {
         self.write(|ctx| {
             let viewport = ctx.viewport();
-            viewport
-                .this_pass
-                .widgets
-                .get(id)
-                .or_else(|| viewport.prev_pass.widgets.get(id))
-                .copied()
+            let rect = viewport.this_pass.widgets.get(id);
+            if let Some(rect) = rect {
+                if rect.rect != Rect::NOTHING {
+                    Some(*rect)
```

Comment:
Maybe add a comment here about why we do this.

Also the fact that we check against `Rect::NOTHING` here is a bit brittle (what if we change the default in… wherever it is set).

Using something like `rect.is_positive() && rect.is_finite()` should be more robust.